src/fhirtypes/ElementDefinition.ts

```diff
@@ -472,6 +475,52 @@ export class ElementDefinition {
     this._original = undefined;
   }
 
+  clearOriginalProperty(pathParts: PathPart[]): void {
+    let currentOriginalElement: any = this._original;
+    // eslint-disable-next-line
+    let currentActualElement: any = this;
+    const clearPath = this.calculateClearPath(pathParts);
+    for (const [i, pathPart] of clearPath.entries()) {
+      if (i < clearPath.length - 1) {
+        const key = pathPart.primitive ? `_${pathPart.base}` : pathPart.base;
+        currentOriginalElement = currentOriginalElement?.[key];
+        currentActualElement = currentActualElement?.[key];
+        const currentIndex = getArrayIndex(pathPart);
+        if (currentIndex != null) {
+          if (Array.isArray(currentOriginalElement)) {
+            currentOriginalElement = currentOriginalElement[currentIndex];
+          }
+          if (Array.isArray(currentActualElement)) {
+            currentActualElement = currentActualElement[currentIndex];
+          }
+        }
+      } else {
+        if (currentOriginalElement?.[pathPart.base] != null) {
+          currentOriginalElement[pathPart.base] = undefined;
+          currentActualElement[pathPart.base] = undefined;
+        }
+        if (pathPart.primitive && currentOriginalElement?.[`_${pathPart.base}`] != null) {
+          currentOriginalElement[`_${pathPart.base}`] = undefined;
+          currentActualElement[`_${pathPart.base}`] = undefined;
+        }
+      }
+    }
+  }
+
+  private calculateClearPath(pathParts: PathPart[]): PathPart[] {
+    const replacementIndex = this._replacementProps.findIndex(
+      replacementPath =>
+        replacementPath.length <= pathParts.length &&
+        replacementPath.every((rp, index) => rp === pathParts[index].base)
+    );
+    if (replacementIndex >= 0) {
+      const clearPath = pathParts.slice(0, this._replacementProps[replacementIndex].length);
+      this._replacementProps.splice(replacementIndex, 1);
```

Comment:
I had to think about this a little as I reviewed it. Might it be helpful to add a couple of comments? For example:
> // Build the clearPath from the actual pathParts since they might have indices or slicenames

and

> // Remove the replacement property from _replacementProps so we don't mistakenly clear it again later